src/middlewared/middlewared/plugins/smb.py

```diff
@@ -22,6 +27,63 @@
 RE_NETBIOSNAME = re.compile(r"^[a-zA-Z0-9\.\-_!@#\$%^&\(\)'\{\}~]{1,15}$")
 
 
+class smbhamode(enum.Enum):
```

Comment:
In python its sort of an unspoken rule to use starting uppercase for class names to not confuse things.

Could we perhaps use: SmbHAMode or something?
Also, should this be an enum.IntEnum?